crates/starknet-devnet/tests/test_v3_transactions.rs

```diff
@@ -0,0 +1,472 @@
+            "--gas-price-fri",
+            "10",
+            "--data-gas-price-fri",
+            "100",
+        ])
+        .await
+        .expect("Could not start Devnet");
+        let (sierra_artifact, casm_hash) = get_simple_contract_in_sierra_and_compiled_class_hash();
+
+        let (signer, account_address) = devnet.get_first_predeployed_account().await;
+
+        let mut account: SingleOwnerAccount<
+            &starknet_rs_providers::JsonRpcClient<starknet_rs_providers::jsonrpc::HttpTransport>,
+            starknet_rs_signers::LocalWallet,
+        > = SingleOwnerAccount::new(
+            &devnet.json_rpc_client,
+            signer,
+            account_address,
+            constants::CHAIN_ID,
+            ExecutionEncoding::New,
+        );
+        account.set_block_id(BlockId::Tag(BlockTag::Latest));
+
+        let sierra_artifact = Arc::new(sierra_artifact);
+
+        transaction_with_less_gas_units_and_or_less_gas_price_should_return_error_or_be_accepted_as_reverted(
+            Action::Declaration(sierra_artifact, casm_hash),
+            Some(&account),
+            Option::<&OpenZeppelinAccountFactory<LocalWallet, JsonRpcClient<HttpTransport>>>::None,
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn deploy_account_happy_path() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+
+        let signer = get_deployable_account_signer();
+        let factory = OpenZeppelinAccountFactory::new(
+            Felt::from_hex_unchecked(CAIRO_0_ACCOUNT_CONTRACT_HASH),
+            constants::CHAIN_ID,
+            signer,
+            devnet.clone_provider(),
+        )
+        .await
+        .unwrap();
+
+        let deploy_v3 = factory.deploy_v3(Felt::THREE);
+        let account_address = deploy_v3.address();
+        devnet
+            .mint_unit(
+                account_address,
+                1e18 as u128,
+                starknet_types::rpc::transaction_receipt::FeeUnit::FRI,
+            )
+            .await;
+
+        let fee_estimate = deploy_v3.estimate_fee().await.unwrap();
+
+        let (gas_units, gas_price) = get_gas_units_and_gas_price(fee_estimate);
+        let result = deploy_v3.gas(gas_units).gas_price(gas_price).send().await.unwrap();
+        assert_tx_successful(&result.transaction_hash, &devnet.json_rpc_client).await;
+    }
+
+    fn get_gas_units_and_gas_price(fee_estimate: FeeEstimate) -> (u64, u128) {
+        let gas_price =
+            u128::from_le_bytes(fee_estimate.gas_price.to_bytes_le()[0..16].try_into().unwrap());
+        let gas_units = fee_estimate
+            .overall_fee
+            .field_div(&NonZeroFelt::from_felt_unchecked(fee_estimate.gas_price));
+
+        (gas_units.to_le_digits().first().cloned().unwrap(), gas_price)
+    }
+    async fn transaction_with_less_gas_units_and_or_less_gas_price_should_return_error_or_be_accepted_as_reverted<
```

Comment:
I notice there's a lot going on in this helper function.

Reply:
[4f2c3e2](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/625/commits/4f2c3e24eaa0e63ac706bdc93b6a9c2252e61ce5)

Reply:
The name was already very descriptive, just wanted to express my doubt about having testing functions with so much if/match logic. The alternative would be having a couple of helper functions to reduce this function's complexity, but I can give my approval without such a refactoring.